

'VB.Net -Exception Handling
'An exception Is a problem that arises during the execution Of a program. An exception Is a response To an exceptional circumstance that arises While a program Is running, such As an attempt To divide by zero.

'Exceptions provide a way To transfer control from one part Of a program To another. VB.Net exception handling Is built upon four keywords: Try, Catch, Finally And Throw.

'Try : A Try block identifies a block of code for which particular exceptions will be activated. It's followed by one or more Catch blocks.

'Catch: A program catches an exception With an exception handler at the place In a program where you want To handle the problem. The Catch keyword indicates the catching Of an exception.

'Finally : The Finally block Is used to execute a given set of statements, whether an exception Is thrown Or Not thrown. For example, If you open a file, it must be closed whether an exception Is raised Or Not.

'Throw: A program throws an exception When a problem shows up. This Is done Using a Throw keyword.

'Syntax
'Assuming a block will raise an exception, a method catches an exception Using a combination Of the Try And Catch keywords. A Try/Catch block Is placed around the code that might generate an exception. Code within a Try/Catch block Is referred To As Protected code, And the syntax For Using Try/Catch looks Like the following

'Try
'    [ tryStatements ]
'    [ Exit Try ]
'[ Catch [ exception [ As type ] ] [ When expression ]
'    [ catchStatements ]
'    [ Exit Try ] ]
'[ Catch ... ]
'[ Finally
'    [ finallyStatements ] ]
'End Try
'You can list down multiple Catch statements To Catch different type Of exceptions In Case your Try block raises more than one exception In different situations.

'Exception Classes In .Net Framework
'In the .Net Framework, exceptions are represented by classes. The exception classes in .Net Framework are mainly directly Or indirectly derived from the System.Exception class. Some of the exception classes derived from the System.Exception class are the System.ApplicationException And System.SystemException classes.

'The System.ApplicationException Class supports exceptions generated by application programs. So the exceptions defined by the programmers should derive from this Class.

'The System.SystemException Class Is the base Class For all predefined system exception.

'The following table provides some Of the predefined exception classes derived from the Sytem.SystemException Class:

'Exception Class Description
'System.IO.IOException	Handles I/O errors.
'System.IndexOutOfRangeException	Handles errors generated when a method refers to an array index out of range.
'System.ArrayTypeMismatchException	Handles errors generated when type Is mismatched with the array type.
'System.NullReferenceException	Handles errors generated from deferencing a null object.
'System.DivideByZeroException	Handles errors generated from dividing a dividend with zero.
'System.InvalidCastException	Handles errors generated during typecasting.
'System.OutOfMemoryException	Handles errors generated from insufficient free memory.
'System.StackOverflowException	Handles errors generated from stack overflow.
'Handling Exceptions
'VB.Net provides a structured solution to the exception handling problems in the form of try And catch blocks. Using these blocks the core program statements are separated from the error-handling statements.

'These error handling blocks are implemented using the Try, Catch And Finally keywords. Following Is an example of throwing an exception when dividing by zero condition occurs



Module Module1

    Sub division(ByVal num1 As Integer, ByVal num2 As Integer)
        Dim result As Integer
        Try
            result = num1 \ num2
        Catch e As DivideByZeroException
            Console.WriteLine("Exception caught: {0}", e)
        Finally
            Console.WriteLine("Result: {0}", result)
        End Try
    End Sub

    Sub Main()
        division(25, 0)
        Console.ReadKey()
    End Sub

End Module


'When the above code Is compiled And executed, it produces the following result:

'Exception caught : System.DivideByZeroException : Attempted to divide by zero. 
'at...
'Result: 0


